### PR TITLE
Update site navigation

### LINKED ISSUE #22: Improve site navigation
I'd love to find a more elegant site navigation solution. Our current solution is robust, however it causes the page to shift.

### PR DESCRIPTION
**Pull request by katydecorah**
I borrowed heavily from: https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-links.html @jsnmrs what do you think? What would you improve?

Fixes #22

### CODE REVIEW

**Comment by jsnmrs** [comment]
@katydecorah looking locally, I'm still seeing a menu flash with the approach. 

This shows two refreshes in Firefox:
nav

I added a commit eliminates the flash. It gives the menu a class of `pre-js` that has a `left: 999em;` and no opacity. When JS initializes that menu, it removes the class.

This JS does a lot — more than I expected. I'm ok with it, if it achieves the functional and visual result you have in mind.

**Comment by katydecorah** [comment]
Per chat, we need to research and understand the keyboard expectations for a site navigation system.

**Comment by jsnmrs** [comment]
@katydecorah It's safe to think about this as a show/hide (supporting links below) — where the button only shows/hides the navigation content:
https://davatron5000.github.io/a11y-nutrition-cards/#disclosure-show-hide

Looking through my links, I realized where I learned about this distinction:
- https://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html
- https://twitter.com/LeonieWatson/status/1042800382882275328

Seeing those cemented it in my mind a few years ago and I haven't needed to produce the rationale since. Now I remember!

For reference, this "Nutrition Card" for menus validates the complex needs that comes with the WAI menu example, completely:
https://davatron5000.github.io/a11y-nutrition-cards/#menu-menu-button

**Comment by katydecorah** [comment]
@jsnmrs those links are awesome! Thank you. That makes a whole lot more sense. Here's what I've done:

1. Removed the added and unneeded aria roles and JavaScript.
2. Added back original JavaScript.
3. Add event listeners to close the menu if you focus off of the menu area or click outside it (curious your thoughts on these!)

Still to do:
- [ ] Improve mobile experience. I think ideally the burger button is always stuck to the top-right.

**Comment by katydecorah** [comment]
I adjusted the flexbox on the site header so I think everything fits and aligns nicely. I also created a `no-js` class and added styles for the navigation in case JS is not enabled.

**Comment by katydecorah** [comment]
Ready for review!

File: scripts/navigation.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+// hiden menu when click is outside the navigation
+document.addEventListener(
+  "click",
+  function (event) {
+    if (!nav.contains(event.target)) {
+      menuHide();
+    }
+  },
+  true
+);
```
Comment (katydecorah) [inline scripts/navigation.js:49]: @jsnmrs I'm particularly interested in your opinion of these two functions. Is there a better way to do it?

Comment (jsnmrs) [reply scripts/navigation.js:49]: That's about as lean as it gets to accomplish outside click and focus catching

**Pull request merged**